Makefile

```diff
@@ -1,10 +1,20 @@
+ARCH=
+CC=gcc
 CFLAGS=-O2 -Wall
 DLLINKFLAGS=-Wl,--kill-at -static-libgcc -shared
 
+ifeq ($(shell uname -m),x86_64)
+ifeq ($(ARCH),i686)
+CC = PATH=/mingw32/bin:$$PATH i686-w64-mingw32-gcc -march=i686
+endif
+endif
+
```

Comment:
Actually, it works. I tested opening a 32-bit shell. BUT, since you said the previous version of this commit worked in a 32-bit SDK, too, _and_ it was a bit easier to underatand/reason about, perhaps we should do as you suggest and revert to that.

FYI: For the MSI (coming on a separate branch), I will need a conditional to compile with the appropriate architecture of gcc. But it'll be simpler han the one shown here (remove the outer conditional and actually assign `ARCH`: `ARCH=$(shell uname -m)`).

I'll make that change. I need this committed before I can implement the `TERMINAL` option in the MSI.